contracts/deploy/base/017_upgrade_amo.js

```diff
@@ -0,0 +1,71 @@
+const { deployOnBaseWithGuardian } = require("../../utils/deploy-l2");
+const addresses = require("../../utils/addresses");
+const { utils } = require("ethers");
+const {
+  deployBaseAerodromeAMOStrategyImplementation,
+} = require("../deployActions");
+
+module.exports = deployOnBaseWithGuardian(
+  {
+    deployName: "017_upgrade_amo",
+  },
+  async ({ ethers }) => {
+    const cOETHbVaultProxy = await ethers.getContract("OETHBaseVaultProxy");
+    const cOETHbVault = await ethers.getContractAt(
+      "IVault",
+      cOETHbVaultProxy.address
+    );
+
+    const cAMOStrategyProxy = await ethers.getContract(
+      "AerodromeAMOStrategyProxy"
+    );
+    const cAMOStrategyImpl =
+      await deployBaseAerodromeAMOStrategyImplementation();
+    const cAMOStrategy = await ethers.getContractAt(
+      "AerodromeAMOStrategy",
+      cAMOStrategyProxy.address
+    );
+
+    return {
+      actions: [
+        {
+          // 1. Upgrade AMO
+          contract: cAMOStrategyProxy,
+          signature: "upgradeTo(address)",
+          args: [cAMOStrategyImpl.address],
+        },
+        {
+          // 2. Reset WETH approvals to 0 on swapRouter and positionManager
+          contract: cAMOStrategy,
+          signature: "safeApproveAllTokens()",
+          args: [],
+        },
+        {
+          // 2. set auto allocate threshold to 1 - gas is cheap
+          contract: cOETHbVault,
+          signature: "setAutoAllocateThreshold(uint256)",
+          args: [utils.parseUnits("0.1", 18)],
```

Comment:
nit: The comment says one, but the argument says 0.1 ether. I assume that 0.1 ether is the correct value, but just want to be sure about it.

Reply:
thanks! corrected